File: teste2.py
import sys

def ler_config(arquivo_config):
    try:
        with open(arquivo_config, "r") as arquivo:
            for linha in arquivo:
                if linha.startswith("num="):
                    return int(linha.strip().split("=")[1])
    except FileNotFoundError:
        print(f"Erro: O arquivo '{arquivo_config}' não foi encontrado.")
        sys.exit(1)
    except ValueError:
        print("Erro: O valor de 'num' no arquivo de configuração não é válido.")
        sys.exit(1)

def main():
    num_esperado = ler_config("config.txt")
    parametros = sys.argv[1:]  

    if len(parametros) != num_esperado:
        print(f"Erro: Esperado {num_esperado} parâmetros, mas recebeu {len(parametros)}.")
        sys.exit(1)

    print(f"Parâmetros recebidos corretamente: {parametros}")

if __name__ == "_main_":
    main()

File: teste2.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int ler_config(const char *arquivo_config) {
    FILE *arquivo = fopen(arquivo_config, "r");
    if (arquivo == NULL) {
        printf("Erro: O arquivo '%s' não foi encontrado.\n", arquivo_config);
        exit(1);
    }

    int num_esperado = -1;
    char linha[100];

    while (fgets(linha, sizeof(linha), arquivo)) {
        if (strncmp(linha, "num=", 4) == 0) {
            num_esperado = atoi(linha + 4);
            break;
        }
    }

    fclose(arquivo);

    if (num_esperado == -1) {
        printf("Erro: O arquivo de configuração não contém um valor válido para 'num'.\n");
        exit(1);
    }

    return num_esperado;
}

int main(int argc, char *argv[]) {
    int num_esperado = ler_config("config.txt");

    if (argc - 1 != num_esperado) {
        printf("Erro: Esperado %d parâmetros, mas recebeu %d.\n", num_esperado, argc - 1);
        return 1;
    }

    printf("Parâmetros recebidos corretamente: ");
    for (int i = 1; i < argc; i++) {
        printf("%s ", argv[i]);
    }
    printf("\n");

    return 0;
}